extensions/auth/openldap/openldap.go

```diff
@@ -0,0 +1,121 @@
+package openldap
+
+import (
+	"fmt"
+
+	management "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/session"
+)
+
+type OLDAPOperations interface {
+	Enable() error
+	Disable() error
+}
+
+const (
+	resourceType = "openldap"
+	schemaType   = "openLdapConfigs"
+)
+
+type OLDAP struct {
+	Config *Config
+
+	session *session.Session
+	client  *rancher.Client
+}
+
+// NewOLDAP constructs OLDAP struct after it reads Open LDAP from the configuration file
+func NewOLDAP(client *rancher.Client, ts *session.Session) (*OLDAP, error) {
+	ldapConfig := new(Config)
+	config.LoadConfig(ConfigurationFileKey, ldapConfig)
+
+	return &OLDAP{
+		Config:  ldapConfig,
+		session: ts,
+		client:  client,
+	}, nil
+}
+
+// Enable is a method of OLDAP, makes a request to the action with the given
+// configuration values
+func (o *OLDAP) Enable() error {
+	var jsonResp map[string]interface{}
+
+	url := o.newActionURL("testAndApply")
+	enableActionInput, err := o.newEnableInputFromConfig()
+	if err != nil {
+		return err
+	}
+
+	err = o.client.Management.Ops.DoModify("POST", url, enableActionInput, &jsonResp)
+	if err != nil {
+		return err
+	}
+
+	o.session.RegisterCleanupFunc(func() error {
+		return o.Disable()
```

Comment:
Why are we calling `return o.Disable()` here?

Reply:
It registers the disable function to the session, in the test if there's a session cleanup called (right now in the teardown) and if the `cleanup` key is set to `true` in the configuration, then it's going to disable it. 

If the `cleanup` key is `false` in the config file, then this disable part is not going to get triggered.